block/config.ml

```diff
@@ -2,9 +2,10 @@ open Mirage
 
 let main = foreign "Unikernel.Main" (console @-> block @-> job)
 
-let img = match get_mode () with
-  | `Xen -> block_of_file "xvda1"
-  | `Unix | `MacOSX -> block_of_file "disk.img"
+let img =
+  if_impl Key.is_xen
+    (block_of_file "xvda1")
+    (block_of_file "disk.img")
```

Comment:
If you really want it, I could implement `add_to_ocamlfind_libraries` and `add_to_opam_packages` again. I removed them because they are quite bad (dependencies should not be added by side effects ...) and imprecise (I prefer having the dependencies at the `foreign` level).